spec/defines/filter/open_tracing_spec.rb

```diff
@@ -3,526 +3,225 @@
-      }
-      it {
-        should contain_file('/etc/repose/open-tracing.cfg.xml').with_content(/service-name=\"test-repose\"/)
-      }
-      it {
-        should contain_file('/etc/repose/open-tracing.cfg.xml').with_content(/token=\"mytoken\"/)
-      }
-      it {
-        should contain_file('/etc/repose/open-tracing.cfg.xml').with_content(/<sampling-probabilistic probability=\"1.0\"/)
-      }
-    end
-
-    context 'with defaults with old namespace' do
-      let(:title) { 'old_namespace_default' }
-      let :pre_condition do
-        "class { 'repose': cfg_new_namespace => false }"
+  on_supported_os.each do |os, os_facts|
+    context "on #{os}" do
+      let(:facts) do
+        os_facts
       end
 
-      let(:params) { {
-        :ensure     => 'present',
-        :service_name => 'test-repose',
-        :http_connection_token    => 'mytoken',
-        :http_connection_endpoint => 'http://example.com/api/traces',
-        :constant_toggle => 'off'
-      } }
-      it {
-        should contain_file('/etc/repose/open-tracing.cfg.xml').
-          with_content(/docs.rackspacecloud.com/)
-      }
-    end
+      context 'default parameters' do
+        let(:title) { 'default' }
+        it {
+          should raise_error(Puppet::Error, /either udp or http connection parameters must be defined/)
+        }
+      end
 
-    context 'with defaults with new namespace' do
-      let(:title) { 'new_namespace_default' }
-      let :pre_condition do
-        "class { 'repose': cfg_new_namespace => true }"
+      context 'with ensure absent' do
+        let(:title) { 'default' }
+        let(:params) { {
+          :ensure => 'absent'
+        } }
+        it {
+          should contain_file('/etc/repose/open-tracing.cfg.xml').with_ensure(
+            'absent')
+        }
       end
 
-      let(:params) { {
-        :ensure     => 'present',
-        :service_name => 'test-repose',
-        :http_connection_token    => 'mytoken',
-        :http_connection_endpoint => 'http://example.com/api/traces',
-        :constant_toggle => 'off'
-      } }
-      it {
-        should contain_file('/etc/repose/open-tracing.cfg.xml').
-          with_content(/docs.openrepose.org/)
-      }
-    end
+      context 'providing http connection - endpoint' do
+        let(:title) { 'connection_http_endpoint' }
+        let(:params) { {
+          :ensure     => 'present',
+          :service_name => 'test-repose',
+          :http_connection_endpoint => 'http://example.com/api/traces',
+          :constant_toggle => 'on'
+        } }
+        it {
+          should contain_file('/etc/repose/open-tracing.cfg.xml').with(
+            'ensure' => 'file',
+            'owner'  => 'repose',
+            'group'  => 'repose',
+            'mode'   => '0660')
+        }
+        it {
+          should contain_file('/etc/repose/open-tracing.cfg.xml').with_content(/connection-http endpoint=\"http:\/\/example.com\/api\/traces\"/)
```

Comment:
for the future, we should merge these with content like other tests instead of writing multiple it tests